photonpump/connection.py

```diff
@@ -1106,6 +1106,35 @@ def __init__(self, connector, dispatcher, credential=None):
     async def __aexit__(self, exc_type, exc, tb):
         await self.close()
 
+    async def find_backwards(self, stream: str, predicate: Callable[[Event], bool]):
+        """
+        Return first event matching predicate or None in none exists.
+
+        Args:
+            stream: The name of the stream.
+            predicate: Callable which get's an event as parameter and return
+            boolean as matching result
+
+        Examples:
+
+            Find event with event.type 'my-event-desired' in stream
+            'my-stream-name'.
+
+            >>> with async.connect() as conn:
+            >>>     event = await conn.find_backwards(
+            >>>         'my-stream-name',
+            >>>         predicate=lambda event: 'my-event-desired' in event.type
+            >>>     )
+        """
+
+        async for event in self.iter(
+            stream, batch_size=10, direction=msg.StreamDirection.Backward
+        ):
+            if predicate(event):
+                return event
+
+        return None
```

Comment:
also here I'm fine with that but not sure what's the standard for our libraries @palankai @epoplavskis like: returning None vs raising an Exception "No event found matching the criteria"?

Reply:
my preference is to raise an exception, though since this is a find function, finding nothing is an option, i would fine with returning none

Reply:
As well why does need to be in photon pump? this cactus functionality cactus specific, not really to be used by other clients.